x-pack/plugins/reporting/public/lib/reporting_api_client/reporting_api_client.test.ts

```diff
@@ -0,0 +1,322 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+jest.mock('moment', () => ({ tz: { guess: jest.fn() } }));
+
+import { tz } from 'moment';
+import { IUiSettingsClient } from 'src/core/public';
+import { httpServiceMock, uiSettingsServiceMock } from 'src/core/public/mocks';
+import { Job } from '../job';
+import { ReportingAPIClient } from './reporting_api_client';
+
+describe('ReportingAPIClient', () => {
+  let uiSettingsClient: jest.Mocked<IUiSettingsClient>;
+  let httpClient: ReturnType<typeof httpServiceMock['createStartContract']>;
```

Comment:
I suggest making this type a bit more explicit:

```suggestion
  let httpClient: jest.Mocked<HttpSetup>;
```